File: tool/py_client/proto/message.py
import urllib
import hashlib
from enum import Enum
from util import *

SIGN1 = "5UY6$f$h"
SIGN2 = "3wokZB%q"
SIGN3 = "%2Fi9TRf"
    
def msgurl(addr, port, version, cmd, msg):
    t = str(timestamp())
    c = str(cmd)
    s = hashlib.md5((SIGN1 + c + SIGN2 + t + SIGN3 + version).encode('utf-8')).hexdigest()
    d = msg.SerializeToString()
    url = "http://%s:%d/msg" % (addr, port)
    postdata = urllib.parse.urlencode({'c': c, 't': t,'d': d, 's': s})
    return url, postdata.encode('utf-8')
    
class LoginCmd(Enum):
    Login=1
    

File: tools/h5client/src/app/proto/message.js
(function () {
    'use strict';

    var md5 = require("blueimp-md5");
    require('sprintf-js');

    module.exports = Message;

    function Message() { }

    var SIGN1 = "5UY6$f$h";
    var SIGN2 = "3wokZB%q";
    var SIGN3 = "%2Fi9TRf";

    Message.msgurl = function (addr, port, version, cmd, msg) {
        var t = String(new Date().getTime());
        var c = String(cmd);
        var s = md5(SIGN1 + c + SIGN2 + t + SIGN3 + version);
        var d = msg.serializeBinary();
        console.log("d1=", d);
        var url = sprintf("http://%s:%s/msg", addr, port);
        var params = { 'c': c, 't': t, 'd': d, 's': s };
        return [url, params];
    };

    Message.posturl = function ($http, url, data, cb_success, cb_fail) {
        $http({
            url: url,
            method: 'POST',
            data: data,
            async: false,
            headers: {
                'Content-Type': 'application/x-www-form-urlencoded',
                "Access-Control-Allow-Origin": "*",
                'Access-Control-Allow-Methods': 'POST',
                'Access-Control-Allow-Headers': 'Accept,X-Custom-Header,X-Requested-With,Content-Type, Origin'
            },
            transformRequest: function (obj) {
                var str = [];
                for (var s in obj) {
                    str.push(encodeURIComponent(s) + "=" + encodeURIComponent(obj[s]));
                    if (s == 'd') {
                        console.log("d2=", encodeURIComponent(obj[s]));
                    }
                }
                return str.join("&");
            }
        }).then(cb_success, cb_fail);
    };

    Message.geturl = function ($http, url, params, cb_success, cb_fail) {
        $http({
            url: url,
            method: 'GET',
            params: params,
            async: false,
            transformRequest: function (obj) {
                var str = [];
                for (var s in obj) {
                    str.push(encodeURIComponent(s) + "=" + encodeURIComponent(obj[s]));
                }
                return str.join("&");
            }
        }).then(cb_success, cb_fail);
    };
})();